Enterprise-CMCS/macpro-mdct-mcr
# CMDCT-2794 (Pt. I): Setup MCR w/ Zustand for State Management

### Description

Converting the state management of MCR from `Context API` to `Zustand` (following the patterns implemented in `MFP`).

This PR covers the initial setup:
- Installation of Zustand in MCR
- Test mocks for Zustand
- `UserProvider` + corresponding tests

### Related ticket(s)


CMDCT-2784

---
### How to test

Running this locally, you should be able to log in / log out from MCR as expected.

**Optionally:** install the [Redux Devtools Chrome extension](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd) or the [Firefox Redux Devtools extension](https://addons.mozilla.org/en-US/firefox/addon/reduxdevtools/)

- Open devtools
- Log in

You should see the `user` object and a few other initial states on the extension ✨ 

### Important updates


`cd services/ui-src yarn install` 

---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
